catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ValidateCommand.java

```diff
@@ -47,178 +55,104 @@
 @Service
 public class ValidateCommand implements Action {
 
-    @Argument(index = 0, name = "fileName", description = "The path to the file that you want to validate", required = true, multiValued = false)
+    @Option(name = "--path", aliases = "-p", description = "The path to the file to be validated")
     @Completion(FileCompleter.class)
-    String filename;
+    String path;
+
+    @Option(name = "--query", aliases = "-q", description = "Text query for catalog entries to validate")
+    String query;
 
     @Reference
     List<MetacardValidator> validators;
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(ValidateCommand.class);
-
-    // Using system out for println is caught by our githooks to prevent bad practice.
-    // We actually need to use it here.
-    protected PrintStream console = System.out;
-
-    // These constants are to help in trying to make the output
-    // both uniform and readable.
-    private static final String INDENT = "|   ";
-
-    private static final String LAST_INDENT = "    ";
-
-    private static final String ELEMENT = "+-- ";
-
-    private static final String LAST_ELEMENT = "`-- ";
-
-    private static final String WARNING_TITLE = "Warnings";
-
-    private static final String ERROR_TITLE = "Errors";
-
-    private static final String WARNING_COLOR = COLOR_CYAN;
+    @Reference
+    CatalogFramework catalog;
 
-    private static final String ERROR_COLOR = COLOR_RED;
+    @Reference
+    FilterBuilder builder;
 
     @Override
     public Object execute() throws Exception {
-
-        if (fileExists() && hasValidators()) {
-            Metacard metacard = createMetacard();
-            if (metacard != null) {
-                runValidators(metacard);
+        int bad = 0;
+        if (validators != null && validators.size() > 0) {
+            List<Metacard> metacards;
+            if (path != null) {
+                metacards = createMetacards();
+            } else if (query != null) {
+                metacards = getMetacardsFromCatalog();
+            } else {
```

Comment:
Similarly, this test can go earlier and fail out the method, simplifying the rest of it.